src/Az.Shared.props

```diff
@@ -19,4 +19,9 @@
     <WarningsAsErrors />
   </PropertyGroup>
 
+  <!-- include shared source code -->
+  <ItemGroup>
+    <Compile Include="$(MSBuildThisFileDirectory)shared\*.cs" />
```

Comment:
I moved this from `Az.props` to here because for some projects like `Authenticators.csproj`, it doesn't depends on `Az.props` but references `Az.Shared.props`.
Although the design has changed and I don't need to reference `ConfigKeys` in `Authenticators`, it's OK to keep this here.